### ISSUE TITLE
Elements with focus should be clearly visible 

### ISSUE THREAD

**Issue by jgiardino** [patternfly/patternfly-next#786]
One of the WCAG items we need to meet is to clearly indicate focus ([WCAG 2.4.7](https://www.w3.org/WAI/WCAG21/quickref/#focus-visible)).

Our original approach has been to rely on the browser defaults for indicating focus. But currently there are a couple of issues with this:

- Our current styles override the default styles provided by Firefox for some elements, like buttons
- Some components might not be able to rely on browser default styles. For example, in a Switch component that's based on a native html checkbox, we're hiding the checkbox which is the element that receives focus, which means the focus is not visible to the user.

P.S. WCAG link above includes links to additional information about how to meet this criteria.

**Comment by LHinson** [comment]
Closing this issue since we have #790
